loadbalancer/zk/publisher.go

```diff
@@ -0,0 +1,84 @@
+package zk
+
+import (
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/loadbalancer"
+	"github.com/go-kit/kit/log"
+	"github.com/samuel/go-zookeeper/zk"
+)
+
+// Publisher yield endpoints stored in a certain zookeeper path. Any kind of
+// change in that path is watched and will update the Publisher endpoints.
+type Publisher struct {
+	client Client
+	path   string
+	cache  *loadbalancer.EndpointCache
+	logger log.Logger
+	quit   chan struct{}
+}
+
+// NewPublisher returns a zookeeper publisher. Zookeeper will start watching the given
+// path for changes and update the Publisher endpoints.
```

Comment:
Can't quite tell in the GitHub interface, but if it's not already true, please wrap all comments at 80col.